### PR TITLE
Added code teaser

### PR DESCRIPTION
**Pull request by jspaaks**
Originally here https://github.com/matchms/matchms-backup/pull/249

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -100,6 +100,43 @@ Install matchms from Anaconda Cloud with
 
   conda install --channel nlesc --channel bioconda --channel conda-forge matchms
 
+Example
+=======
+
+Below is a small example of using matchms to calculate the Cosine score between mass Spectrums.
```
Comment (fdiblen) [inline README.rst]:
```suggestion
Below is a small example of using matchms to calculate the Cosine score between mass Spectrums. It uses [pesticides.mgf](https://github.com/matchms/matchms/blob/d2858874096524f6be1f12e613178a9ada96aac8/tests/pesticides.mgf) dataset.
```

File: README.rst [hunk 2]
```diff
@@ -100,6 +100,43 @@ Install matchms from Anaconda Cloud with
...
+    from matchms.importing import load_from_mgf
+    from matchms.filtering import default_filters
+    from matchms.filtering import normalize_intensities
+    from matchms import calculate_scores
+    from matchms.similarity import CosineGreedy
+
+    file = load_from_mgf("tests/pesticides.mgf")
+
+    def my_filter(spectrum):
+        '''Clean and enhance the spectrums with a filter'''
```
Comment (fdiblen) [inline README.rst]:
```suggestion
        '''Clean and enhance the spectrums with matchms filters.'''
```

File: README.rst [hunk 3]
```diff
@@ -100,6 +100,43 @@ Install matchms from Anaconda Cloud with
...
+        spectrum = default_filters(spectrum)
```
Comment (fdiblen) [inline README.rst:113]:
```suggestion
        spectrum = default_filters(spectrum) # see `matchms/filtering/default_filters.py` for the list of filters 
```

File: README.rst [hunk 4]
```diff
@@ -100,6 +100,43 @@ Install matchms from Anaconda Cloud with
...
+
+.. code-block:: python
+
+    from matchms.importing import load_from_mgf
+    from matchms.filtering import default_filters
+    from matchms.filtering import normalize_intensities
+    from matchms import calculate_scores
+    from matchms.similarity import CosineGreedy
+
+    file = load_from_mgf("tests/pesticides.mgf")
```
Comment (fdiblen) [inline README.rst:106]:
```suggestion
    # load mgf dataset
    file = load_from_mgf("tests/pesticides.mgf")
```

File: README.rst [hunk 5]
```diff
@@ -100,6 +100,43 @@ Install matchms from Anaconda Cloud with
...
+
+    def my_filter(spectrum):
```
Comment (fdiblen) [inline README.rst]:
```suggestion
    # define a function which will apply some filters
    def my_filter(spectrum):
```

File: README.rst [hunk 6]
```diff
@@ -100,6 +100,43 @@ Install matchms from Anaconda Cloud with
...
+        '''Clean and enhance the spectrums with a filter'''
+        spectrum = default_filters(spectrum)
+        spectrum = normalize_intensities(spectrum)
```
Comment (fdiblen) [inline README.rst:115]:
```suggestion
        spectrum = normalize_intensities(spectrum) # a filter to normalize intensities
```

File: README.rst [hunk 7]
```diff
@@ -100,6 +100,43 @@ Install matchms from Anaconda Cloud with
...
+        return spectrum
+
+    spectrums = [my_filter(spectrum) for spectrum in file]
```
Comment (fdiblen) [inline README.rst]:
```suggestion
    # apply filters to each spectrum in the dataset
    spectrums = [my_filter(spectrum) for spectrum in file]
```

File: README.rst [hunk 8]
```diff
@@ -100,6 +100,43 @@ Install matchms from Anaconda Cloud with
...
+
+    similarity_function = CosineGreedy()
```
Comment (fdiblen) [inline README.rst]:
```suggestion
    # calculate similarity using cosine greedy function # see matchms/similarity/CosineGreedy.py
    similarity_function = CosineGreedy()
```

File: README.rst [hunk 9]
```diff
@@ -100,6 +100,43 @@ Install matchms from Anaconda Cloud with
...
+    scores = calculate_scores(spectrums, spectrums, similarity_function)
```
Comment (fdiblen) [inline README.rst]:
```suggestion
   # calculate the scores for filtered spectrums using the similarity function we picked
    scores = calculate_scores(spectrums, spectrums, similarity_function)
```

File: README.rst [hunk 10]
```diff
@@ -100,6 +100,43 @@ Install matchms from Anaconda Cloud with
...
+
+    for score in scores:
```
Comment (fdiblen) [inline README.rst:125]:
```suggestion
    # show the calculated scores
    for score in scores:
```

File: readthedocs/index.rst [hunk 11]
```diff
@@ -6,14 +6,51 @@
...
-Vector representation of mass spectral data (e.g. MS-MS) and spectral similarity measure.
+Python library for fuzzy comparison of mass spectrum data and other Python objects.
 
 .. toctree::
    :maxdepth: 3
    :caption: Contents:
 
    API <api/matchms.rst>
 
+Example
```
Comment (fdiblen) [inline readthedocs/index.rst:17]: Can you add the same comments to this code example as well?

**Review by fdiblen — CHANGES_REQUESTED** [review]
I have suggested few minor changes. I think it is also good to have comments in code examples.

File: README.rst (hunk 4 above)
Comment (sverhoeven) [reply README.rst:106]: Comment does not tell me anything more then the code, not adding

File: README.rst (hunk 6 above)
Comment (sverhoeven) [reply README.rst:115]: Comment does not tell me anything more then the code, not adding

File: README.rst (hunk 3 above)
Comment (sverhoeven) [reply README.rst:113]: I would like to link to the implementation, but I think it distracts too much from the example.

File: README.rst (hunk 5 above)
Comment (sverhoeven) [reply README.rst]: Docstring of my_filter already says this

File: README.rst (hunk 7 above)
Comment (sverhoeven) [reply README.rst]: Same as before, comment does not explain code better

File: README.rst (hunk 8 above)
Comment (sverhoeven) [reply README.rst]: Same as before, comment does not explain code better

File: README.rst (hunk 7 above)
Comment (sverhoeven) [reply README.rst]: We could also replace the my_filter and list comprehension with a for loop.

File: README.rst (hunk 1 above)
Comment (sverhoeven) [reply README.rst]: Solved it slightly different, is it OK?

File: README.rst (hunk 3 above)
Comment (sverhoeven) [reply README.rst:113]: Added link to docs, docs will need to be improved to be useful.

File: README.rst (hunk 5 above)
Comment (sverhoeven) [reply README.rst]: my_filter is gone

File: README.rst (hunk 6 above)
Comment (sverhoeven) [reply README.rst:115]: Comment `Scale peak intensities to maximum of 1` added

File: README.rst (hunk 9 above)
Comment (sverhoeven) [reply README.rst]: Comment added with link to docs. Doc for CosineGreedy does not exist see #49

**Comment by sverhoeven** [comment]
Switched from my_filter and list comprehension to for loop. Added comments to each section and method. @fdiblen can you have a look again?

File: README.rst (hunk 1 above)
Comment (fdiblen) [reply README.rst]: Good. Thanks

**Review by fdiblen — APPROVED** [review]
Thank you for the changes. Looks very good.

**Pull request merged**